contracts/toCKB-typescript/src/switch/preterm_redeem/mod.rs

```diff
@@ -1,7 +1,144 @@
+        || input_toCKB_data.x_lock_address.as_ref() != out_toCKB_data.x_lock_address.as_ref()
+        || input_toCKB_data.signer_lockscript.as_ref() != out_toCKB_data.signer_lockscript.as_ref()
+    {
+        return Err(Error::InvariantDataMutated);
+    }
+    Ok(lot_size)
+}
+
+pub fn verify_burn(lot_size: u128, out_toCKB_data: &ToCKBCellDataView) -> Result<(), Error> {
+    let mut deposit_requestor = false;
+    let mut input_sudt_sum: u128 = 0;
+    let mut output_sudt_sum: u128 = 0;
+    let mut input_index = 0;
+    loop {
+        let cell_type = load_cell_type(input_index, Source::Input);
+        match cell_type {
+            Err(SysError::IndexOutOfBound) => break,
+            Err(_err) => panic!("iter input return an error"),
+            Ok(cell_type) => {
+                if !(cell_type.is_some()
+                    && cell_type.unwrap().code_hash().raw_data().as_ref()
+                        == SUDT_CODE_HASH.as_ref())
+                {
+                    input_index += 1;
+                    continue;
+                }
+
+                let lock = load_cell_lock(input_index, Source::Input)?;
+
+                if lock.as_slice() == out_toCKB_data.redeemer_lockscript.as_ref() {
+                    deposit_requestor = true;
+                }
+
+                let data = load_cell_data(input_index, Source::Input)?;
+                let mut buf = [0u8; 16];
+                if data.len() == 16 {
+                    buf.copy_from_slice(&data);
+                    input_sudt_sum += u128::from_le_bytes(buf)
+                }
+                input_index += 1;
+            }
+        }
+    }
+
+    let mut output_index = 0;
+    loop {
+        let cell_type = load_cell_type(output_index, Source::Output);
+
+        match cell_type {
+            Err(SysError::IndexOutOfBound) => break,
+            Err(_err) => panic!("iter output return an error"),
+            Ok(cell_type) => {
+                if !(cell_type.is_some()
+                    && cell_type.unwrap().code_hash().raw_data().as_ref()
+                        == SUDT_CODE_HASH.as_ref())
+                {
+                    output_index += 1;
+                    continue;
+                }
+                let data = load_cell_data(output_index, Source::Output)?;
+                let mut buf = [0u8; 16];
+                if data.len() == 16 {
+                    buf.copy_from_slice(&data);
+                    output_sudt_sum += u128::from_le_bytes(buf)
+                }
+                output_index += 1;
+            }
+        }
+    }
+
+    if deposit_requestor && output_sudt_sum != 0 {
+        return Err(Error::XTBurnInvalid);
+    }
+    if !deposit_requestor {
+        let signer_fee: u128 = lot_size * SIGNER_FEE_RATE.0 / SIGNER_FEE_RATE.1;
+        if (input_sudt_sum != lot_size + signer_fee) || (output_sudt_sum != signer_fee) {
+            return Err(Error::XTBurnInvalid);
+        }
+    }
```

Comment:
ref: https://github.com/nervosnetwork/axon-internal/issues/84#issuecomment-685431371

we should consider the change issue